rsocket-core/src/commonMain/kotlin/io/rsocket/kotlin/core/ReconnectableRSocket.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2015-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.rsocket.kotlin.core
+
+import io.ktor.utils.io.core.*
+import io.rsocket.kotlin.*
+import io.rsocket.kotlin.payload.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.flow.*
+
+internal typealias ReconnectPredicate = suspend (cause: Throwable, attempt: Long) -> Boolean
+
+@Suppress("FunctionName")
+@OptIn(ExperimentalCoroutinesApi::class, FlowPreview::class)
+internal fun ReconnectableRSocket(
```

Comment:
Why internal? Experimental and Preview seems enough.

Reply:
But why we need it public?

Reply:
The example you give shows it being used directly, I probably just missed the entry point and better example.